src/aggregation/bucket/term_agg.rs

```diff
@@ -544,6 +548,27 @@ impl SegmentTermCollector {
                 let val = bool::from_u64(val);
                 dict.insert(IntermediateKey::Bool(val), intermediate_entry);
             }
+        } else if self.column_type == ColumnType::IpAddr {
+            let arc_inner = agg_with_accessor
+                .accessor
+                .values
+                .clone()
+                .downcast_arc::<CompactSpaceU64Accessor>()
```

Comment:
I'm surprised downcasting logic is required here. Are there really no better way?

Reply:
The main problem here is with regards to generics.
We need a codec specific method on a specific trait variant `ColumnValues<u64>`.

We could return `ColumnValuesU64Ext` instead of `ColumnValues<u64>`, which would include the method

```rust 

trait U64ColumnValuesExt: ColumnValues<u64> {
    compact_to_u128(&self, compact: u32) -> u128 {
        unimplemented!() // unimplemented for all expect compact space
    }
}
```

Having codec specific methods on a common trait is a little bit weird.
Casting is also not great, but having the mechanism may also allow some performance gains by enabling inlining in some cases.

In both cases we will need to know the underlying codec before calling the codec specific method.